Sources/Sentry/SentryProfiler.mm

```diff
@@ -155,7 +157,7 @@
     }
 
     {
-        std::lock_guard<std::mutex> l(_gSamplesArrayLock);
+        std::lock_guard<std::mutex> l(_gDataStructureLock);
```

Comment:
Ouch, ok let me get a test written for this.

Reply:
Wrote a test that reproduced this behavior in https://github.com/getsentry/sentry-cocoa/pull/3018/commits/a51cf976b5ccfaf9d7cd2c54a020d2003de3c961 and fixed in the next commit 👍🏻

Reply:
Note that only the threadMetadata structure still had this issue, because it has a mutable subdictionary, where the copy of the top-level structure still referenced the same mutable instance that could later be modified. The solution was to do a copy one level deeper. None of the other data structures had this issue, so the calls to `copy` of the top-level structures is sufficient.